### DISCUSSION TITLE
Encoding issue encountered on a particular beatmap

### DISCUSSION THREAD

**Discussion by jem-suu** [ppy/osu#18574] [Q&A]
### Type

Cosmetic

### Bug description

Not exactly sure why it's only affecting this [particular beatmap](https://osu.ppy.sh/beatmapsets/798691#osu/1677157), but I have `Prefer metadata in original language` enabled, and it seems to be displaying the song title of this beatmap on the selection menu on the side in gibberish, instead of hangul. Song info panel and the in-game beatmap info detail screen render the hangul correctly, and other songs with hangul metadata render correctly, so I'm not sure if this is isolated to this particular beatmap set or not.

### Screenshots or videos

![osu!_2022-06-04_22-03-08](https://user-images.githubusercontent.com/49844693/172025528-447eac48-95b4-4f9a-830c-01b385320028.png)
![osu!_2022-06-04_22-04-19](https://user-images.githubusercontent.com/49844693/172025530-a225949f-797a-4407-b98a-9951167cd476.png)
_(Pixelated sections are from the discord overlay as I was in a call at the time)_

### Version

2022.528.0

### Logs

Not Applicable

**Comment by frenzibyte** [comment]
I couldn't reproduce on neither macOS nor Windows. Can you try clearing your font cache and see if that makes the title render correctly?
 1. open osu! folder using settings, then close game.
 2. go to `cache` folder, 
 3. rename the `fonts` folder inside to anything else.
 4. run the game back and check affected beatmap.

**Reply by jem-suu** [reply]
Apologies if it took a long time to answer, think I had to reinstall lazer after dealing with some messed up situation of somehow having the files of three different versions on my PC.

However, I was able to reproduce the issue on a fresh install after having reimported my beatmap collection from legacy osu! and the same issue is persistant, but now the beatmap information on the left develops the issue. In-game beatmap info still works fine, cause I'm assuming it's data is pulled from the web, not from local storage

![osu!_2022-06-05_22-39-17](https://user-images.githubusercontent.com/49844693/172071856-753a3b69-be12-4aa9-8172-d99c7f77fa4c.png)
![osu!_2022-06-05_22-43-53](https://user-images.githubusercontent.com/49844693/172071858-0ab6a6b6-ccee-4dc2-9399-ab640edcbe55.png)

I then removed and redownload the affected beatmap through lazer, and it seems to have no issues with metadata

![image](https://user-images.githubusercontent.com/49844693/172071952-97de581d-76d5-4cf9-a4d3-ac967c822d25.png)

Thus, I'm probably assuming that this issue comes through migration from legacy osu!, rather than something wrong with that specific beatmap.

**Reply by smoogipoo** [reply]
Have you done the above suggestion...? It's very specific in what you need to do, and not the same as reinstalling osu!.

**Reply by peppy** [reply]
I don't think it's texture cache related as you can see every glyph has the correct padding (the `oe` character is especially telling).

@jem-suu can you also export the beatmap via the editor (File > Export..) and share it here?

**Reply by bdach** [reply]
It's not the texture cache. The actual data is being decoded using the incorrect encoding.

The raw bytes in hex are:

```
EB 9A 9C EB 91 90 EB 9A 9C EB 91 90
```

If the above bytes are decoded to utf8, the correct result will be shown. If they are decoded with [windows-1252](https://en.wikipedia.org/wiki/Windows-1252) then the garbage shown on the screenshots will be decoded.

This may be an issue with *something* .NET trying to autodetect encoding and spectacularly failing. Not sure, haven't tried reproducing on windows yet. I don't expect to see this on other platforms because mac and linux have mostly reached sanity and generally don't bother with anything but utf8.

**Reply by peppy** [reply]
@bdach I'm presuming you are checking the bytes in the actual beatmap file.

Question then is why this works on import (online metadata lookup succeeding where it doesn't during user's stable import..?)

I'll take a look into this. We should likely be forcing utf8 for decoding beatmap files, as that is the expectation.

**Reply by peppy** [reply]
Weirdly, we *are* forcing utf8:

https://github.com/ppy/osu/blob/6944151486e677bfd11f2390163aca9161defbbf/osu.Game/IO/LineBufferedReader.cs#L22

**Reply by peppy** [reply]
@jem-suu was your legacy import done a long time ago? ie. potentially before 2020-03-31? If so, that's likely going to be the issue.

See https://github.com/ppy/osu/commit/f6f5de7ad16fd416bdbb7ad11fa205be085f66c6, where the UTF8 parsing was actually enforced.
